lib/SIL/TypeLowering.cpp

```diff
@@ -2520,3 +2520,75 @@ TypeConverter::getContextBoxTypeForCapture(ValueDecl *captured,
   
   return boxType;
 }
+
+static void countNumberOfInnerFields(size_t &fieldsCount, SILModule &Module,
+                                     SILType Ty) {
+  if (auto *structDecl = Ty.getStructOrBoundGenericStruct()) {
+    if (!structDecl->hasFixedLayout(Module.getSwiftModule(),
+                                    ResilienceExpansion::Minimal)) {
+      return;
```

Comment:
Perhaps this should be an assertion instead, because FSO should not be trying to explode resilient (ie address-only) types at all.